src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -16,14 +16,24 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
         public bool MayBeNull => State == NullableFlowState.MaybeNull;
         public bool IsNotNull => State == NullableFlowState.NotNull;
 
-        public static bool CanContainNull(TypeSymbol type)
+        private static bool CanContainNull(TypeSymbol type)
         {
             return type is null || !type.IsValueType || type.IsNullableTypeOrTypeParameter();
         }
 
-        public static TypeWithState ForType(TypeSymbol type) => new TypeWithState(type, CanContainNull(type) ? NullableFlowState.MaybeNull : NullableFlowState.NotNull);
+        public static TypeWithState ForType(TypeSymbol type)
+        {
+            var state = CanContainNull(type) ? NullableFlowState.MaybeNull : NullableFlowState.NotNull;
```

Comment:
Should these two checks be done in the constructor for `TypeWithState`?

Reply:
Perhaps although that would mean passing in a state that is then compared with `MaybNull`. It seemed slightly cleaner to avoid check from this factory method.

---
In reply to: [271059294](https://github.com/dotnet/roslyn/pull/34630#discussion_r271059294) [](ancestors = 271059294)